core/src/main/scala/com/salesforce/op/stages/impl/feature/SmartTextVectorizer.scala

```diff
@@ -130,13 +131,22 @@ class SmartTextVectorizer[T <: Text](uid: String = UID[SmartTextVectorizer[T]])(
 
     // build metadata describing output
     val shouldTrackNulls = $(trackNulls)
+    val shouldTrackLen = $(trackTextLen)
     val unseen = Option($(unseenName))
 
     val categoricalColumns = if (categoricalFeatures.nonEmpty) {
       makeVectorColumnMetadata(shouldTrackNulls, unseen, smartTextParams.categoricalTopValues, categoricalFeatures)
     } else Array.empty[OpVectorColumnMetadata]
     val textColumns = if (textFeatures.nonEmpty) {
-      makeVectorColumnMetadata(textFeatures, makeHashingParams()) ++ textFeatures.map(_.toColumnMetaData(isNull = true))
+      if (shouldTrackLen) {
+        makeVectorColumnMetadata(textFeatures, makeHashingParams()) ++
```

Comment:
can this be DRYed out?

Reply:
Hmmm, best I can come up with is
```
val textColumns = if (textFeatures.nonEmpty) {
      
      makeVectorColumnMetadata(textFeatures, makeHashingParams()) ++
        (if (shouldTrackLen) textFeatures.map(_.toColumnMetaData(descriptorValue =
          OpVectorColumnMetadata.TextLenString)) else Array.empty[OpVectorColumnMetadata]) ++
        (if (shouldTrackNulls) textFeatures.map(_.toColumnMetaData(isNull = true))
      else Array.empty[OpVectorColumnMetadata])
      
    } else Array.empty[OpVectorColumnMetadata]
```

which looks less readable to me...